src/structured_data.rs

```diff
@@ -96,6 +96,29 @@ impl StructuredData {
         DataIdentifier::Structured(self.name, self.type_tag)
     }
 
+    /// Delete self
+    pub fn delete_self_against_successor(&mut self,
+                                         other: &StructuredData)
+                                         -> Result<(), RoutingError> {
+        match self.validate_self_against_successor(other) {
```

Comment:
Maybe this should verify that `other` is empty, too, at least `data` and `current_owner_keys`.
(Not sure how realistic that is, but if an attacker were able to intercept a client's valid `Post` request, they could turn it into a valid `Delete` otherwise, without having to change the signature.)